fix(ci): reloader OCIRepository ref.tag contains digest suffix, breaking flux-local helm template

## Problem

Every PR opened against `main` currently shows a red `Flux Local - Test` check. The failure is not caused by the PR under review — it is caused by a pre-existing bad shape in the `reloader` `OCIRepository` manifest, introduced as a side-effect of enabling `pinDigests: true` in #2813.

### The failure

From the `Flux Local - Test` CI log (reproducible on any open PR, e.g. #2827):

```
FAILED kubernetes/cluster0/flux::cluster-apps-reloader::kube-system/reloader
flux_local.exceptions.HelmException: Command 'helm template reloader
  oci://ghcr.io/stakater/charts/reloader --namespace kube-system ...
  --version 2.2.11@sha256:05c912e1ed0703618454bf75a61e5ee19789f1108dd8ccf1e9fd246e4d127469 ...'
  failed with return code 1
Error: improper constraint: 2.2.11@sha256:05c912e1ed0703618454bf75a61e5ee19789f1108dd8ccf1e9fd246e4d127469
```

### Root cause

In `kubernetes/cluster0/apps/kube-system/reloader/app/helm-release.yaml`, the `OCIRepository.spec.ref.tag` field combines a semver tag with a digest:

```yaml
ref:
  tag: 2.2.11@sha256:05c912e1ed0703618454bf75a61e5ee19789f1108dd8ccf1e9fd246e4d127469
```

- Flux's `source-controller` tolerates this hybrid form (which is why the cluster itself is healthy).
- `flux-local` (the CI validator) invokes `helm template --version <ref.tag>`, and Helm's `--version` flag only accepts pure semver constraints. Helm rejects the `@sha256:...` suffix as an "improper constraint" and the template step fails.

Per the [Renovate Flux manager docs](https://docs.renovatebot.com/modules/manager/flux/#ocirepository-support):

> The `flux` manager only updates `OCIRepository` fields that have a `digest` or `tag` key. If `tag` *and* `digest` exist then `digest` takes precedence.

The Flux-native and Renovate-supported shape is two separate fields:

```yaml
ref:
  tag: \"2.2.11\"
  digest: sha256:05c912e1ed0703618454bf75a61e5ee19789f1108dd8ccf1e9fd246e4d127469
```

This shape makes both `flux-local`/`helm template` happy (clean semver in `--version`) and Flux's source-controller happy (digest verification preserved).

### Why did Renovate write the broken shape?

Issue #2813 enabled `pinDigests: true` on the `docker` datasource. For conventional container images (`image.repository` + `image.tag` pairs inside Helm values), Renovate correctly appends `@sha256:...` to the `tag` string — container runtimes accept that combined form, so this is fine. For Flux `OCIRepository` resources, Renovate should write to the separate `ref.digest` field instead, but there is a known open bug:

- [renovatebot/renovate#42082](https://github.com/renovatebot/renovate/issues/42082) — \"Correctly specify the `digest` field when performing digest pinning\" (open)

So Renovate currently mishandles digest pinning for `OCIRepository.ref` and writes the hybrid `tag@sha256:...` string. Flux accepts it silently; `helm template` does not.

## Scope

### Files that MUST be fixed (in scope)

Only `OCIRepository` resources are affected. A repo-wide search (`rg '@sha256:' kubernetes --type yaml`) finds two:

1. **`kubernetes/cluster0/apps/kube-system/reloader/app/helm-release.yaml`** (line 14)
   - `OCIRepository` consumed by a `HelmRelease` → templated by `flux-local` → **currently breaks CI**.
2. **`kubernetes/cluster0/flux/config/flux.yaml`** (line 11)
   - `OCIRepository` consumed by a `Kustomization` (flux-manifests bootstrap) → not templated by helm → **does not break CI today**, but same shape issue. Fix for consistency and future-proofing (e.g. if `flux-local` ever validates `OCIRepository.ref.tag` as semver regardless of consumer).

### Files that MUST NOT be changed (out of scope, already correct)

All other `@sha256:` occurrences in `kubernetes/` are in legitimate container image reference contexts where the combined form is valid:

- **Helm values `image.tag` fields** (~24 occurrences) — e.g. `kubernetes/cluster0/apps/media/sonarr/app/helm-release.yaml`, `.../home/zigbee2mqtt/app/helm-release.yaml`, etc. These are opaque strings passed through to pod specs; container runtimes accept `tag@sha256:...`. Leave as-is.
- **Kustomize `images[].newTag`** — `kubernetes/cluster0/apps/system-upgrade/system-upgrade-controller/app/kustomization.yaml` (line 10). Kustomize passes this through to container refs. Leave as-is.
- **Plain `image:` fields in workload specs** — `kubernetes/cluster0/apps/system-upgrade/system-upgrade-controller/plans/{server,agent}-plan.yaml`. Already use the `image: name@sha256:...` form without a separate tag field. Leave as-is.

### Renovate configuration changes

Likely **none required** for the manifest fix itself — Renovate's Flux manager will recognise the split `ref.tag` + `ref.digest` shape and update both fields correctly going forward (per the docs quoted above).

However, two defensive options should be considered:

1. **Add a custom manager comment** on each `OCIRepository` to pin datasource/versioning explicitly (belt-and-braces). Not strictly needed — include only if you want explicit Renovate annotations.
2. **Monitor next reloader chart bump**. If Renovate regresses and re-introduces the hybrid `tag@sha256:...` shape due to [renovatebot/renovate#42082](https://github.com/renovatebot/renovate/issues/42082), we will need a `packageRules` entry scoped to the Flux manager + OCIRepository that disables digest pinning on `ref.tag` — or we switch to a `customManager` with explicit regex for these two files. **Do not pre-emptively add this rule** — wait to see actual behaviour on the next bump to avoid over-engineering.

No changes to `.github/renovate.json5`, `.github/renovate/automerge.json5`, or `.github/renovate/labels.json5` are expected as part of this fix. If the worker agent determines a rule *is* needed after manual testing or investigation, they should call it out rather than silently adding one.

## Acceptance criteria

### Primary (must pass)

- [ ] **AC1**: `kubernetes/cluster0/apps/kube-system/reloader/app/helm-release.yaml` has its `OCIRepository.spec.ref` split into two fields: `tag: \"2.2.11\"` and `digest: sha256:05c912e1ed0703618454bf75a61e5ee19789f1108dd8ccf1e9fd246e4d127469` (exact values preserved from the current combined string).
- [ ] **AC2**: `kubernetes/cluster0/flux/config/flux.yaml` has its `OCIRepository.spec.ref` split into two fields: `tag: \"v2.8.5\"` and `digest: sha256:ffb44e1d6ade00aed4b0451a217cdfd3025ffed71137c2c1bb77f44d1b7f36f8` (exact values preserved).
- [ ] **AC3**: No other files under `kubernetes/` are modified. Specifically, no `image.tag` fields inside Helm values, no Kustomize `newTag`, and no plain `image:` fields are touched.
- [ ] **AC4**: After the PR is opened, `Flux Local - Test` passes green on the PR branch.
- [ ] **AC5**: `Flux Local - Diff (helmrelease)` and `Flux Local - Diff (kustomization)` show no material change to rendered manifests for reloader or flux-manifests (the digest is preserved, so the image Flux pulls is identical).

### Secondary (should pass)

- [ ] **AC6**: Commit message follows repo convention (`fix(ci): ...` or `chore(ci): ...` with a clear one-line summary and a body explaining the hybrid `tag@sha256:` → split `tag`+`digest` change).
- [ ] **AC7**: PR description references this issue with `Closes #<this-issue-number>` and links to the Renovate docs excerpt and upstream bug [renovatebot/renovate#42082](https://github.com/renovatebot/renovate/issues/42082) for posterity.
- [ ] **AC8**: No Renovate config files are modified unless the worker has a justified reason (documented in the PR description).

### Post-merge verification (follow-up, not blocking PR merge)

- [ ] **AC9**: After merge, confirm the next Renovate run does not re-introduce the hybrid `tag@sha256:` shape. If it does, open a follow-up issue to add a Renovate `packageRules` workaround or `customManager` for `OCIRepository` resources.
- [ ] **AC10**: Flux reconciles the reloader HelmRelease and flux-system Kustomization without error (`kubectl get helmrelease reloader -n kube-system` and `kubectl get kustomization flux -n flux-system` both `Ready=True`).

## References

- Failing run on #2827: https://github.com/prismatic-koi/home-ops/actions/runs/24529868570/job/71709983652
- Origin of the hybrid shape: #2813 (enabled `pinDigests: true`)
- Renovate Flux manager docs: https://docs.renovatebot.com/modules/manager/flux/#ocirepository-support
- Upstream Renovate bug: https://github.com/renovatebot/renovate/issues/42082
- Flux `OCIRepository` ref spec: https://fluxcd.io/flux/components/source/ocirepositories/#artifact

--- a/.github/renovate.json5
+++ b/.github/renovate.json5
@@ -52,6 +52,22 @@
       ],
       allowedVersions: '!/-(armhf|arm64|armv7|amd64|linux)$/',
     },
+    {
+      // Flux manager + OCIRepository hybrid digest shape is broken:
+      // Renovate writes `tag: <semver>@sha256:<digest>` instead of populating
+      // the separate `ref.digest` field (renovatebot/renovate#42082). flux-local
+      // then passes the hybrid string (or the bare digest, when split) to
+      // `helm template --version`, which Helm rejects as a non-semver
+      // constraint, breaking CI.
+      //
+      // Disable digest pinning on the reloader OCIRepository specifically.
+      // The flux-manifests OCIRepository keeps digest pinning because it is
+      // consumed by a Kustomization (not templated by helm) and is unaffected.
+      // See #2831 for full context.
+      matchManagers: ['flux'],
+      matchFileNames: ['kubernetes/cluster0/apps/kube-system/reloader/app/helm-release.yaml'],
+      pinDigests: false,
+    },
   ],
   customManagers: [
     {

--- a/kubernetes/cluster0/apps/kube-system/reloader/app/helm-release.yaml
+++ b/kubernetes/cluster0/apps/kube-system/reloader/app/helm-release.yaml
@@ -11,7 +11,7 @@ spec:
     mediaType: application/vnd.cncf.helm.chart.content.v1.tar+gzip
     operation: copy
   ref:
-    tag: 2.2.11@sha256:05c912e1ed0703618454bf75a61e5ee19789f1108dd8ccf1e9fd246e4d127469
+    tag: "2.2.11"
   url: oci://ghcr.io/stakater/charts/reloader
 ---
 # yaml-language-server: $schema=https://kubernetes-schemas.pages.dev/helm.toolkit.fluxcd.io/helmrelease_v2.json

--- a/kubernetes/cluster0/flux/config/flux.yaml
+++ b/kubernetes/cluster0/flux/config/flux.yaml
@@ -8,7 +8,8 @@ spec:
   interval: 10m
   url: oci://ghcr.io/fluxcd/flux-manifests
   ref:
-    tag: v2.8.6@sha256:45ed09e8241e4b0ee38d0fc8d8e42b6002d55c880b0859c9c484bf22340116d5
+    tag: "v2.8.6"
+    digest: sha256:45ed09e8241e4b0ee38d0fc8d8e42b6002d55c880b0859c9c484bf22340116d5
 ---
 apiVersion: kustomize.toolkit.fluxcd.io/v1
 kind: Kustomization

